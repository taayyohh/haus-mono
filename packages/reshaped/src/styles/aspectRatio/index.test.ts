import getAspectRatioStyles from "./index";

describe("Styles/AspectRatio", () => {
	test("handles positive value", () => {
		expect(getAspectRatioStyles(1 / 2)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(getAspectRatioStyles(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getAspectRatioStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getAspectRatioStyles({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});
