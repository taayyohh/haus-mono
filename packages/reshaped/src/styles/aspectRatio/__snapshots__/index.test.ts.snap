// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/AspectRatio > handles 0 value 1`] = `null`;

exports[`Styles/AspectRatio > handles positive value 1`] = `
{
  "classNames": "_root_618290",
  "variables": {
    "--rs-ratio-s": 0.5,
  },
}
`;

exports[`Styles/AspectRatio > handles responsive value 1`] = `
{
  "classNames": "_root_618290",
  "variables": {
    "--rs-ratio-l": 2,
    "--rs-ratio-m": 0,
    "--rs-ratio-s": 4,
  },
}
`;

exports[`Styles/AspectRatio > handles undefined value 1`] = `null`;
